### PR TITLE
Added FAST documentation

### PR DESCRIPTION
**Pull request by pentschev**

### CODE REVIEW

**Comment by gpryor** [comment]
Nice!

File: include/af/image.h [hunk 1]
```diff
@@ -82,6 +82,21 @@ AFAPI void grad(array& rows, array& cols, const array& in);
 
 AFAPI array regions(const array& in, af::connectivity connectivity=AF_CONNECTIVITY_4, dtype type=f32);
 
+/**
+    C++ Interface
+
+    A circle of radius 3 pixels, translating into a total of 16 pixels, is checked for sequential segments of pixels much brighter or much darker than the central one. For a pixel p to be considered a feature, there must exist a sequential segment of arc_length pixels in the circle around it such that all are greather than (p + thr) or smaller than (p - thr). After all features in the image are detected, if nonmax is true, the non-maximal suppression is applied, checking all detected features and the features detected in its 8-neighborhood and discard it if its score is non maximal.
```
Comment (9prady9) [inline include/af/image.h]: @pentschev  All common descriptions about the functions(C/C++ API) are being held in one single place `docs\details\image.dox`. Please refer to `colorspace` & `af_colorspace` functions documentation to get an idea on what needs to be changed. If you still have questions, we can talk about it on Monday.

**Comment by pentschev** [comment]
@9prady9 please check commit 13ccc7e, should be in the correct place now.

**Comment by 9prady9** [comment]
:+1:

**Pull request merged**